src/Core/Authorization/AuthorizationResolver.cs

```diff
@@ -232,7 +232,15 @@ public void SetEntityPermissionMap(RuntimeConfig runtimeConfig)
                 bool isStoredProcedureEntity = entity.Source.Type is EntitySourceType.StoredProcedure;
                 if (isStoredProcedureEntity)
                 {
-                    SupportedHttpVerb[] methods = entity.Rest.Methods;
+                    SupportedHttpVerb[] methods;
+                    if (entity.Rest.Methods is not null)
+                    {
+                        methods = entity.Rest.Methods;
+                    }
+                    else
+                    {
+                        methods = (entity.Rest.Enabled) ? new SupportedHttpVerb[] { SupportedHttpVerb.Post } : Array.Empty<SupportedHttpVerb>();
+                    }
```

Comment:
1. True, `entity.Rest.Methods` will get processed during config deserialization. With the current logic, `entity.Rest.Methods` will never come out to be `null`.

The code branch where `entity.Rest.Methods` can be `null` is added
a) To avoid using null forgiving operator
b) To safeguard against exceptions in the future, - if the deserialization logic should change in a way that the entity.Rest.Methods could come to be `null`.


2) Having an entity with entity.Rest disabled while having rest methods populated does not break. In `SqlMetadataProvider`, the logic for populating `EntityPathToEntityName` takes care of checking if the rest is enabled before mapping a certain path to an entity. 

When a REST request is made, `EntityPathToEntityName` is checked to see if the `path` in the request is a known path (maps to any of the configured entities or not). If an entity is not exposed through REST endpoint, this data structure would not contain a mapping for the `path` and would lead to failure of the request (404 not found).